### ISSUE TITLE
Contact fields

### ISSUE THREAD

**Issue by switzersc** [codefordenver/rmmfi#32]
As a RMMFI member want to be able to track community member's roles so I can easily keep track of people and generate reports.
- [x] Update fields on contact model to match spreadsheet
- [ ] Organize visually on contact page
- [x] Be sure to include Volunteer types/states and ratings on Sheet 2 of the google spreadsheet

https://docs.google.com/spreadsheets/d/1UCwrql1lihq9ZeThEaX9M8oSWkx27y72zIWhy1pbzXo/edit#gid=2104256254

**Comment by wdoug** [comment]
Potential volunteer, active volunteer, potential donor, active donor, misc., etc

**Comment by brettevans** [comment]
Confirm with RMMFI what fields will be needed, and populated from where; survey, manual, etc.

**Comment by brettevans** [comment]
Find existing fields document.

**Comment by brettevans** [comment]
Existing fields document here:
https://docs.google.com/spreadsheets/d/1iyTDjKrQkvG3qNuo0r-1G02K6o1BtQUFDc2vXjZj4lY/edit#gid=1653254995

**Comment by KeynesYouDigIt** [comment]
I added  a guess as to how this should look to the fields doc under sheet two to emphasize high level fields (like prosp volunteer v donor, etc) that deeply affect contact requirements

LMK if that helps at all! we should run it by Brendan before finalizing fields.

Edit: I am requesting edit access now if thats OK

**Comment by brettevans** [comment]
Vince, see if you can edit here: https://docs.google.com/spreadsheets/d/1UCwrql1lihq9ZeThEaX9M8oSWkx27y72zIWhy1pbzXo/edit#gid=38464013

**Comment by KeynesYouDigIt** [comment]
thanks!

**Comment by jenshurley** [comment]
##Linking Contact fields to bootcamp cycles 
Reference: Bootcamp Process document 
In reference to this statement in particular --"Would be great if this process could happen per cycle without losing information of past bootcamp processes"(at top) -- I have a few questions and ideas about how to make this  happen.  https://drive.google.com/file/d/0B_M4SZHaH5j1QW5nZG9PZkJsUE0/view

The relationship of bootcamps to contacts has come up in creating contact fields that capture information necessary to run meaningful Roster, #38 Prospective and #39 Ready reports. 

###Are bootcamps entered as Event objects? (Or maybe Opportunities? Or Product?)

If so, it would be nice to have a lookup-type field (the same kind that like Account is for the Contact object) for the bootcamp with which the contact was most recently associated. 

I started making contact fields to show availability for and past involvement in each bootcamp, and I had some questions about how and where to store that information. 

My current plan is to 
.create a field set for each Bootcamp class ('cycle'?)
..Availability field (Boolean - checkbox)
..Status (Picklist (or Boolean for each?) for Invited, Confirmed, No Response etc -- exact wording in Volunteer Attributes spreadsheet Brett linked above) 
..(and other fields that are specific to a bootcamp)
.create a few contact layouts (one showing field sets for all the bootcamps, and one just showing most recent)
.assign field dependencies (e.g. if Available checked, Status field appears) 
.try adding add new Event for a few bootcamps and assign Bootcamp fields to Lookup fields

**Comment by KeynesYouDigIt** [comment]
That all makes sense, I would move forward with that. 

IMHO,
If status is not boolean, we would have to write a rule for something to change automatically, otherwise the manual process could become difficult to keep track of. I think "Availability" and "Status" should be boolean to "yes/no" and "invited/responded", with a third boolean of "declined/accepted" triggered by a status of "responded" if possible. 

LMK if you need any help, see yal monday!

**Comment by jenshurley** [comment]
Agreed, Vince. I'll create those fields as you specify. Next step (probably new card) is defining validation rules.  
https://help.salesforce.com/HTViewHelpDoc?id=workflow_field_update_considerations.htm&language=en_US might help here

**Comment by haydenevans** [comment]
Added Contacted? field (Boolean/Checkbox) for keeping track of which contacts have been contacted. Included as filter logic in "Prospective Volunteer" Report. Need a field for survey completion linked to surveys to complete this report.

**Comment by jenshurley** [comment]
Changed fields for easy mapping of Lead to Contact - 
LEAD | CONTACT
Contacted? | Contacted as lead
Interested in clients 
Interested in org 
Interested roles
